src/dc_mimeparser.rs

```diff
@@ -277,14 +266,16 @@ pub unsafe fn dc_mimeparser_parse(
                 dc_trim(subj);
                 if 0 != *subj.offset(0isize) {
                     for part in mimeparser.parts.iter_mut() {
-                        if part.type_0 == 10i32 {
+                        if part.type_0 == Viewtype::Text {
+                            let msg_c = part.msg.as_ref().unwrap().strdup();
```

Comment:
I think cases where we have the pattern that we start from a rust string, need to create a temporary C string to call an unsafe function should be handled with `let tmp = CString::new().unwrap(); tmp.as_ptr()` because that's more robust than the `.strdup(); free()` variation as it leaves more memory management to rust.

But in this case `new_txt` is only used to be converted back to `String` so would it not be better to change the `dc_mprintf()` to `format!()`?

Reply:
Replacing `dc_mprintf` with `format!`, when format string contains EN-DASH somewhy caused problems. I did not found why. See my other PR.

But ideally, yes -- `dc_mprintf` should be replaced with `format!`